client/src/main/java/com/aliyun/emr/rss/client/compress/RssZstdCompressor.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aliyun.emr.rss.client.compress;
+
+import java.util.zip.CRC32;
+import java.util.zip.Checksum;
+
+import com.github.luben.zstd.Zstd;
+
+public class RssZstdCompressor extends RssZstdTrait implements Compressor {
+  private final int compressionLevel;
+  private final Checksum checksum;
+  private byte[] compressedBuffer;
+  private int compressedTotalSize;
+
+  public RssZstdCompressor() {
+    this(256 * 1024);
+  }
+
+  public RssZstdCompressor(int blockSize) {
+    this(blockSize, 1);
+  }
+
+  public RssZstdCompressor(int blockSize, int level) {
+    compressionLevel = level;
+    checksum = new CRC32();
+    initCompressBuffer(blockSize);
```

Comment:
Do you mean use the max boundary size of compressed data to replace the fixed configured blocksize when initialize compressor?
I think we would check if current buffer size could meet the compression requirement when each time compress, if not then it will re-new a buffer, it follows the same logic with lz4 compressor.
```
    int maxDestLength = (int) Zstd.compressBound(length);
    if (compressedBuffer.length - HEADER_LENGTH < maxDestLength) {
      initCompressBuffer(maxDestLength);
    }
```